boolean/boolean.py

```diff
@@ -1003,14 +1009,70 @@ def symbols(*args):
     "(": 20,
 }
 
+ANDS = ('*', '&', 'and')
+ORS = ('+', '|', 'or')
+NOTS = ('~', '!', 'not')
+PARENS = ('(', ')')
+OPS = ANDS + ORS + NOTS + PARENS
+ZERO_OR_ONE = ('0', '1')
+
 
-def parse(expr, eval=True):
+def tokenizer(expr):
     """
-    Returns a boolean expression created from the given string.
+    Return an iterable of tokens as tuples of (is_symbol, token, start row,
+    start column) given an `expr`.
+    `is_symbol` must be True if the token is a symbol. 
+    
+    If `is_symbol` the token must of an operation token or and opening '(' or
+    closing parenthesis ')'.
+
+    Recognized operations tokens are:
+    - for and:  '*', '&', 'and'
+    - for or: '+', '|', 'or'
+    - for not: '~', '!', 'not'
     """
     if not isinstance(expr, basestring):
         raise TypeError("Argument must be string but it is %s." % type(expr))
 
+    ignored_token_types = (
+        tokenize.NL, tokenize.NEWLINE, tokenize.COMMENT,
+        tokenize.INDENT, tokenize.DEDENT,
+        tokenize.ENDMARKER
+    )
+
+    OPS = ANDS + ORS + NOTS + PARENS
+
+    # note: unbalanced expression may raise a TokenError here.
+    tokens = tokenize.generate_tokens(StringIO(expr).readline)
+
+    for toktype, tok, (srow, scol,), _, _ in tokens:
+        if toktype in ignored_token_types or not tok.strip():
+            continue
+
+        is_symbol = False
+        if tok.lower() in OPS:
+            is_symbol = False
+            tok = tok.lower()
+        elif toktype == tokenize.NAME or tok in ZERO_OR_ONE:
+            is_symbol = True
+        else:
+            raise TypeError("Unknown token: %(tok)s at line: %(srow)d, column: %(scol)d" % locals())
+
+        yield is_symbol, tok, srow, scol
```

Comment:
About the tokenizer class, the class method here would only return `self.AND`, `self.OR`, etc. objects or string, instead of always strings, also making `is_symbol` unneeded. The parser would then receive a tokenizer object or a tokenizer class or a string, would create a tokenizer object if it isn't one already, and could then check directly for those tokens as `if tok is tokenizer.AND:` ... `else:  # it's a symbol` In its loop.